src/rulesets-new/json-api-rules/__tests__/resource-object-rules.test.ts

```diff
@@ -0,0 +1,312 @@
+import { OpenAPIV3 } from "@useoptic/openapi-utilities";
+import { RuleRunner, TestHelpers } from "@useoptic/rulesets-base";
+import { context } from "../../__tests__/fixtures";
+import { resourceObjectRules } from "../resource-object-rules";
+
+const baseJson = TestHelpers.createEmptySpec();
+
+describe("resource object rules", () => {
+  describe("valid GET responses", () => {
+    test("status code 200", () => {
+      const afterJson = {
+        ...baseJson,
+        paths: {
+          "/api/example": {
+            get: {
+              responses: {
+                "200": {
+                  description: "",
+                  content: {
+                    "application/vnd.api+json": {
+                      schema: {
+                        type: "object",
+                        properties: {
+                          data: {
+                            type: "object",
+                            properties: {
+                              id: {
+                                type: "string",
+                                format: "uuid",
+                              },
+                              type: {
+                                type: "string",
+                              },
+                            },
+                          },
+                          links: {
+                            properties: {
+                              self: {
+                                type: "string",
+                              },
+                            },
+                          },
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      } as OpenAPIV3.Document;
+
+      const ruleRunner = new RuleRunner([resourceObjectRules]);
+      const ruleInputs = {
+        ...TestHelpers.createRuleInputs(baseJson, afterJson),
+        context,
+      };
+      const results = ruleRunner.runRulesWithFacts(ruleInputs);
+      expect(results.length).toBeGreaterThan(0);
+      expect(results.every((result) => result.passed)).toBe(true);
+      expect(results).toMatchSnapshot();
+    });
+
+    test("status code 201", () => {
+      const afterJson = {
+        ...baseJson,
+        paths: {
+          "/api/example": {
+            get: {
+              responses: {
+                "201": {
```

Comment:
Hmm... a GET shouldn't respond 201 Created. We might be missing a rule. If this is pre-existing behavior, fine for a followup.

Reply:
It looks like we're missing a similar one to this for GET + PATCH responses https://github.com/snyk/sweater-comb/blob/main/src/rulesets/jsonapi.ts#L56-L74